apps/nextjs/next.config.mjs

```diff
@@ -1,22 +1,16 @@
 // @ts-check
-import { env } from "./src/env/server.mjs";
 import withTM from "next-transpile-modules";
 
 /**
- * Don't be scared of the generics here.
- * All they do is to give us autocompletion when using this.
- *
- * @template {import('next').NextConfig} T
- * @param {T} config - A generic parameter that flows through to the return type
- * @constraint {{import('next').NextConfig}}
+ * Run `build` or `dev` with `SKIP_ENV_VALIDATION` to skip env validation.
+ * This is especially useful for Docker builds or linting.
  */
-function defineNextConfig(config) {
-  return config;
-}
+!process.env.SKIP_ENV_VALIDATION && (await import("./src/env/server.mjs"));
```

Comment:
this makes is so you can do stuff like `SKIP_ENV_VALIDATION=1 pnpm lint` and it won't care about the env-validation.

Reply:
Nice, very helpful comment as well.